lib/modules/dosomething/dosomething_campaign/includes/Campaign.php

```diff
@@ -444,6 +455,49 @@ protected function getMobileCommonsData() {
     ];
   }
 
+  protected function getPartners() {
+    $data = [];
+
+    // @TODO: only accounts for single partner at the moment.
+    // Update extract function to grab array of partners instead of just first id.
+    $partner_data_collection_ids = dosomething_helpers_extract_field_data($this->node->field_partners);
+
+    if (! $partner_data_collection_ids) {
+      return $data;
+    }
+
+    if (! is_array($partner_data_collection_ids)) {
+      $partner_data_collection_ids = [$partner_data_collection_ids];
+    }
+
+    foreach ($partner_data_collection_ids as $id) {
+      $partner = [];
+
+      $data_collection = field_collection_item_load($id);
+      $data_taxonomy = taxonomy_term_load(dosomething_helpers_extract_field_data($data_collection->field_partner));
+
+      $partner['name'] = $data_taxonomy->name;
+      $partner['sponsor'] = (bool) dosomething_helpers_extract_field_data($data_collection->field_is_sponsor);
+      $partner['copy'] = dosomething_helpers_extract_field_data($data_collection->field_partner_copy);
+      $partner['uri'] = dosomething_helpers_extract_field_data($data_collection->field_partner_url);
+
+      $logo = dosomething_helpers_extract_field_data($data_taxonomy->field_partner_logo);
+
+      if ($logo) {
+        $partner['media'] = [
+          'uri' => image_style_url('wmax-423px', $logo['uri']),
```

Comment:
Yeah, but it's already what we use in the web app. On the DS we use CSS to help us tame the size of partner logos. I referenced the image "style" used in the ds_taxonomy module.